src/download.py

```diff
@@ -46,6 +47,9 @@ def download(args: Namespace):
         # supply the API with the auth credentials  
         # this is only needed for download not search
         api.set_auth_details(args.cred)
+        # check if the download_dir is exist
+        if args.download_dir and not os.path.exists(args.download_dir):
+            os.makedirs(args.download_dir)
```

Comment:
This fragment is duplicated three times in the file, it's better to move it out to the outer scope.

`download_dir` is of type Path, and a default value is used in the tmp directory if the user didn't provide a value which means the `download_dir` will always be provided.
https://github.com/muhammads97/SGLICollect/blob/ba716af6df1d7cebccbadf32d0291de30ae979d4/src/args.py#L100-L105

I would recommend moving this modification to the `args.py` file which handles most of the arguments, after parsing the arguments the directory can be created as follows:
```suggestion
        args.download_dir.mkdir(exist_ok=True, parents=True)
```